web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount.tsx

```diff
@@ -166,7 +169,9 @@ export const CreditsAmount = ({
   useEffect(() => {
     if (maxCreditsSelected) {
       setValue(CREDITS_AMOUNT, creditsAvailable, { shouldValidate: true });
-      setValue(CURRENCY_AMOUNT, spendingCap, { shouldValidate: true });
+      setValue(CURRENCY_AMOUNT, spendingCap as number, {
```

Comment:
maybe we could just do `if (maxCreditsSelected && spendingCap)` instead of type assertion here?

Reply:
In this case should be alright because I don't think `spendingCap` is ever going to be anything other than a number at the time this is executed. That said, it's probably more robust to do `if (maxCreditsSelected && spendingCap)` as you suggest, thanks!